File: golang/removeNthNodeFromEndOfList.go
func removeNthFromEnd(head *ListNode, n int) *ListNode {
    if head.Next == nil {
        return nil 
    }
	fastPoint := head
	for ; 0 < n && fastPoint.Next != nil; n-- {
		fastPoint = fastPoint.Next
	}
	
	if n != 0 {
		return head.Next
	}
	slowPoint := head

	for fastPoint.Next != nil {
		fastPoint = fastPoint.Next
		slowPoint = slowPoint.Next
	}
	slowPoint.Next = slowPoint.Next.Next

	return head
}

File: python3/removeNthNodeFromEndOfList.py
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        if head.next is None:
            return None

        fast_point = head

        while 0 < n and fast_point.next is not None:
            fast_point = fast_point.next
            n -= 1

        if n != 0:
            return head.next
        slow_point = head

        while fast_point.next is not None:
            fast_point = fast_point.next
            slow_point = slow_point.next

        slow_point.next = slow_point.next.next
        return head
